onnxruntime/contrib_ops/cuda/math/bias_softmax.cc

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "bias_softmax.h"
+#include "core/providers/common.h"
+#include "core/providers/cuda/cudnn_common.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace cuda {
+
+#define REGISTER_KERNEL_TYPED(T)                                                \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(                                                \
+      BiasSoftmax,                                                              \
+      kMSDomain,                                                                \
+      1,                                                                        \
+      T,                                                                        \
+      kCudaExecutionProvider,                                                   \
+      KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<T>()), \
+      BiasSoftmax<T>);                                                              
+
+template <typename T>
+Status BiasSoftmax<T>::ComputeInternal(OpKernelContext* ctx) const {
+
+  typedef typename ToCudaType<T>::MappedType CudaT;
+
+  auto X_data = reinterpret_cast<const CudaT*>(ctx->Input<Tensor>(0)->template Data<T>());
+  const TensorShape& X_shape{ctx->Input<Tensor>(0)->Shape()};
+  auto B_data = reinterpret_cast<const CudaT*>(ctx->Input<Tensor>(1)->template Data<T>());
+  const TensorShape& B_shape{ctx->Input<Tensor>(1)->Shape()};
+  auto Y_data = reinterpret_cast<CudaT*>(ctx->Output(0, X_shape)->template MutableData<T>());
+
+  const int64_t softmax_axis = HandleNegativeAxis(softmax_axis_, X_shape.NumDimensions());
+  int N = (int)X_shape.SizeToDimension(softmax_axis);
```

Comment:
is there a need to cast to int? if so, avoid C-style casts, per https://google.github.io/styleguide/cppguide.html#Casting